### ISSUE TITLE
Initialising neuron state with some array or distribution

### ISSUE THREAD

**Issue by arvoelke** [nengo/nengo#1415]
It would be nice to be able to specify either an array or a distribution to initialise the state of the neurons. Is there currently any way to do this (even something hacky)? For the case of a LIF population this would mean initialising the membrane voltage. For adaptive LIF, one might also want to initialise the adaptive state.

The motivation for this is, an initial state of all zeros is somewhat 'pathological' (for lack of a better word), that is:
 - Given a constant input, all of the neurons will tend to burst together. This tendency continues until the state of all neurons effectively become independent of one another. Statisticians refer to this as the "mixing time" of a stochastic process. This is especially noticeable when the firing rates are small, for example, below when `max_rates=nengo.dists.Uniform(10, 20)`:
![low_rates](https://user-images.githubusercontent.com/5420057/36813693-653119ce-1ca3-11e8-92a7-d454f1c606f1.png)
 - This state may not be plausible in some models, in that the only way to "get back" to an initial state of all zeros is to inhibit the population. Even no stimulation is still representing zero with active neurons via bias currents, unless the tuning curves are sparse like in a thresholding population. In many of our models, most of the ensembles may never return to this initial state. This means that, until the mixing time is reached, the probe data is, technically speaking, a statistical anomaly.

The best solution I currently have is to run long continuous simulations, and ignore the first few seconds of probe data (i.e., treat it as some "transient").

From the perspective of Nengo, the `neuron_type` could accept some extra parameters to initialise the state. But reasonable defaults (other than all zeros) are difficult. One approach that I think would be very nice is to use the tuning curves and evaluation points (and assume something reasonable about how the input varies between sampling points) to derive a distribution for the state at any point in time. Then, default to sampling this distribution to initialise the state. (This is something that I have been working on / thinking about, and so I would be happy to know if there is already any work in this direction. There are also some interesting connections here to spike-time chaos and Denève-style networks.)

**Comment by drasmuss** [comment]
Related to #316

This would be the hacky way to do it (obviously not a good interface, but just if you want to get it working):
```
import nengo
import numpy as np

with nengo.Network(seed=0) as net:
    a = nengo.Ensemble(10, 1)
    p = nengo.Probe(a.neurons)

with nengo.Simulator(net) as sim:
    sim.signals[sim.model.sig[a.neurons]['voltage']] = np.random.randn(10)
    sim.run_steps(5)
    print(sim.data[p])
```

**Comment by arvoelke** [comment]
Oh awesome! This will keep me happy/busy for a while. :) Thanks for the snippet!

**Comment by arvoelke** [comment]
I've added a helper function to `nengolib` (currently accessible under the branch `initialize_lif` via https://github.com/arvoelke/nengolib/pull/156) that can be used as follows:

```python
from nengolib.neurons import initialize_lif

# ... build model ...

with nengo.Simulator(model) as sim:
    initialize_lif(sim, x)

    # ... run model ...
```

to correctly initialize the `voltage` and `refractory_time` of the `LIF` ensemble `x`.

This is the "mathematically correct" thing to do for `LIF` neurons, assuming that the state of all synapses are being initialized to `0`. Specifically the state is initialized such that each neuron is uniformly distributed within its inter-spike interval. This is equivalent to what happens when the simulation is fast-forwarded to some sufficiently large `t`. Note that this is different (but may be approximated by) the voltage being uniformly distributed.

**Before**:

![uninitialized](https://user-images.githubusercontent.com/5420057/49478980-7eb78180-f7ef-11e8-9106-c684955a32d0.png)

**After**:

![initialized](https://user-images.githubusercontent.com/5420057/49478982-7f501800-f7ef-11e8-8a9c-31844f97ed6e.png)

The code is in the linked PR, and the above plots are generated via:

```
py.test nengolib/tests/test_neurons.py --plots=plots
```

**Comment by arvoelke** [comment]
Closed by https://github.com/arvoelke/nengolib/pull/156. Example usage can be found in [docs/master/nengolib.neurons.init_lif](https://arvoelke.github.io/nengolib-docs/master/nengolib.neurons.init_lif.html#nengolib.neurons.init_lif), to be released in upcoming v0.4.3. Similar feature requests are welcome.
